src/components/dev-dot-content/index.tsx

```diff
@@ -1,13 +1,94 @@
 import { ReactElement } from 'react'
 import classNames from 'classnames'
+import { MDXProvider } from '@mdx-js/react'
+import { MDXRemote } from 'next-mdx-remote'
+import ImageConfig from 'components/image-config'
 import { DevDotContentProps } from './types'
+import {
+	MdxA,
+	MdxOrderedList,
+	MdxUnorderedList,
+	MdxListItem,
+	MdxTab,
+	MdxTabs,
+	MdxTable,
+	MdxH1,
+	MdxH2,
+	MdxH3,
+	MdxH4,
+	MdxH5,
+	MdxH6,
+	MdxP,
+	MdxInlineCode,
+	MdxBlockquote,
+	MdxTryHcpCallout,
+	MdxTip,
+	MdxHighlight,
+	MdxNote,
+	MdxWarning,
+	MdxCodeBlockConfig,
+	MdxCodeTabs,
+	MdxPre,
+} from './mdx-components'
 import s from './dev-dot-content.module.css'
 
+const DEFAULT_MDX_COMPONENTS = {
+	// native html elements
+	a: MdxA,
+	blockquote: MdxBlockquote,
+	h1: MdxH1,
+	h2: MdxH2,
+	h3: MdxH3,
+	h4: MdxH4,
+	h5: MdxH5,
+	h6: MdxH6,
+	inlineCode: MdxInlineCode,
+	li: MdxListItem,
+	ol: MdxOrderedList,
+	p: MdxP,
+	pre: MdxPre,
+	table: MdxTable,
+	ul: MdxUnorderedList,
+
+	// custom primitives
+	CodeBlockConfig: MdxCodeBlockConfig,
+	CodeTabs: MdxCodeTabs,
+	Highlight: MdxHighlight,
+	ImageConfig,
+	Note: MdxNote,
+	Tab: MdxTab,
+	Tabs: MdxTabs,
+	Tip: MdxTip,
+	TryHcpCallout: MdxTryHcpCallout,
+	Warning: MdxWarning,
+}
+
+/**
+ * Handles providing custom MDX components to MDXRemote. To provide additional
+ * components, use the `mdxRemoteProps` .
+ *
+ * @TODO remove the `children` prop after replacing the 1 instance
+ * https://app.asana.com/0/1202097197789424/1203820006759167/f
+ */
 const DevDotContent = ({
 	className,
 	children,
+	mdxRemoteProps,
 }: DevDotContentProps): ReactElement => {
-	return <div className={classNames(s.root, className)}>{children}</div>
+	const shouldRenderChildren = Boolean(children)
+
+	let content
+	if (shouldRenderChildren) {
+		content = children
+	} else {
+		content = (
+			<MDXProvider components={DEFAULT_MDX_COMPONENTS}>
+				<MDXRemote {...mdxRemoteProps} />
+			</MDXProvider>
```

Comment:
[suggestion] I still would like us to remove the `<MDXProvider>` usage altogether here. Passing components as a prop to `<MDXRemote>` will have the same behavior and will remove an extra layer. 🧹

Reply:
I tried that before pushing things up for review, but found the code to be quite a bit harder to read because there was a lot of destructuring & spreading going on. Could you reply with a little snippet of what you have in mind, and then maybe we can jam on it and see if we can come up with a balance?

Reply:
How does this look?
```suggestion
			<MDXRemote
				{...mdxRemoteProps}
				components={{ ...DEFAULT_MDX_COMPONENTS, ...mdxRemoteProps?.components }} 
			/>
```

Reply:
Okay first of all... my brain completely forgot I could pull out the `components` prop separately from the other props spreading. 🤣 

Second of all... I both love and hate that JavaScript lets you spread a property from an object even if that property is undefined. :lolsob:

![image](https://user-images.githubusercontent.com/43934258/214711776-5d9a760a-5351-4357-9619-a344cb4d0873.png)

This is great!

Reply:
I'm pushing up a commit now for these changes because prettier wanted the lines broken up differently than your suggestion 😁

Reply:
Here it is: https://github.com/hashicorp/dev-portal/pull/1565/commits/bc07db780f9b87025b3315f12206dbf2d4cba2b5

Reply:
Hehe I tried my best to appease prettier....maybe next time 😂